src/Domain/CoordinateMaps/TimeDependent/ShapeMapTransitionFunctions/RegisterDerivedWithCharm.hpp

```diff
@@ -0,0 +1,8 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+namespace domain::CoordinateMaps {
+void register_derived_with_charm() noexcept;
```

Comment:
this either needs a more specific name or namespace, or should be moved up into the `CoordinateMaps/` directory. Else it can collide with a similar function in `CoordinateMaps`. Perhaps a `domain::CoordinateMaps::ShapeMapTransitionFunctions` namespace would be most appropriate?